File: week12/23/prob3_Battleships_in_Board/prob3.c
#include <stdio.h>

int countBattleships(char** board, int boardSize, int* boardColSize) {
    int count = 0;

    for (int i = 0; i < boardSize; i++) {
        for (int j = 0; j < boardColSize[i]; j++) {
            // Count only the top-left corner of each battleship
            if (board[i][j] == 'X') {
                if (i > 0 && board[i-1][j] == 'X') continue; // Check above
                if (j > 0 && board[i][j-1] == 'X') continue; // Check left
                count++;
            }
        }
    }
    return count;
}

// Example usage
int main() {
    char* board[] = {
        "X..X",
        "...X",
        "...X"
    };
    int boardColSize[] = {4, 4, 4};
    printf("%d\n", countBattleships(board, 3, boardColSize)); // Output: 2
    return 0;
}

File: week12/23/prob3_Battleships_in_Board/prob3.py
def countBattleships(board):
    m, n = len(board), len(board[0])
    count = 0

    for i in range(m):
        for j in range(n):
            # Count only the top-left corner of each battleship
            if board[i][j] == 'X':
                if i > 0 and board[i-1][j] == 'X':  # Check above
                    continue
                if j > 0 and board[i][j-1] == 'X':  # Check left
                    continue
                count += 1
    return count

# Example usage
board = [
    ['X', '.', '.', 'X'],
    ['.', '.', '.', 'X'],
    ['.', '.', '.', 'X']
]
print(countBattleships(board))  # Output: 2
